### PR TITLE
[SYSTEMDS-3185] Federated Lookup Table Statistics

### PR DESCRIPTION
**Pull request by ywcb00**
Hi,
this PR (1) adds statistics regarding the federated lookup table to the federated statistics, (2) splits the federated PUT request count into separate counts for the individual Object types (e.g., Matrix, Frame, Scalar, and MatrixCharacteristics) and adds a respective byte count for matrices and frames, and (3) refactors the whole Statistics class and moves related statistics into their own statistics class (Codegen, HOP, Native, ParamServ, ParFor, Spark, Transform).

Thanks for review :)

### CODE REVIEW

File: src/main/java/org/apache/sysds/api/DMLScript.java [hunk 1]
```diff
@@ -83,35 +83,36 @@
 
 public class DMLScript 
 {
-	private static ExecMode   EXEC_MODE          = DMLOptions.defaultOptions.execMode;           // the execution mode
```
Comment (j143) [inline src/main/java/org/apache/sysds/api/DMLScript.java:86]: Hi, you can always create a new PR with formatting changes!

Comment (ywcb00) [reply src/main/java/org/apache/sysds/api/DMLScript.java:86]: Should I create a separate PR for the formatting changes next time?

File: src/main/java/org/apache/sysds/api/DMLScript.java [hunk 2]
```diff
@@ -461,7 +462,9 @@ public static void setGlobalFlags(DMLConfig dmlconf) {
 		if(DMLScript.GPU_MEMORY_UTILIZATION_FACTOR < 0) {
 			throw new RuntimeException("Incorrect value (" + DMLScript.GPU_MEMORY_UTILIZATION_FACTOR + ") for the configuration:" + DMLConfig.GPU_MEMORY_UTILIZATION_FACTOR);
 		}
-		
+
+		DMLScript.USE_LOCAL_SPARK_CONFIG |= dmlconf.getBooleanValue(DMLConfig.USE_LOCAL_SPARK_CONFIG);
```
Comment (ywcb00) [inline src/main/java/org/apache/sysds/api/DMLScript.java:466]: @mboehm7 Is this resolving the problem with the tests you mentioned in #1421 ?

File: src/test/java/org/apache/sysds/test/functions/federated/multitenant/FederatedMultiTenantTest.java [hunk 3]
```diff
@@ -104,11 +104,10 @@ public void testSumSameWorkersSP() {
...
 
-//FIXME still runs into blocking
-//	@Test
-//	public void testSumSharedWorkersSP() {
-//		runMultiTenantSharedWorkerTest(OpType.SUM, 3, 9, ExecMode.SPARK);
-//	}
+	@Test
+	public void testSumSharedWorkersSP() {
+		runMultiTenantSharedWorkerTest(OpType.SUM, 3, 9, ExecMode.SPARK);
+	}
```
Comment (ywcb00) [inline src/test/java/org/apache/sysds/test/functions/federated/multitenant/FederatedMultiTenantTest.java:110]: @mboehm7 I wasn't able to reproduce the error you mentioned. Is the blocking state of this test caused by the not set _DMLScript.USE_LOCAL_SPARK_CONFIG_ flag or is there another problem? :thinking:

File: src/main/java/org/apache/sysds/api/DMLScript.java (hunk 1 above)
Comment (j143) [reply src/main/java/org/apache/sysds/api/DMLScript.java:86]: Generally, changing formatting is a bit confusing for the reviewers, and difficult to check for accidental typos. I think it's okay for this PR. :)

> An example: https://github.com/apache/systemds/commit/bf4ba16b9aaa9afee20a3f1c03b0ff49c5346a9d

Comment (ywcb00) [reply src/main/java/org/apache/sysds/api/DMLScript.java:86]: Alright, thank you for the feedback. I will try to isolate the formatting changes from the actual PR next time. :)

**Comment by phaniarnab** [comment]
Thanks for the refactoring @ywcb00. 
I did a few syntax changes (split long lines into two) and renamed a couple of new files while merging.

**Pull request closed without merging**